### ISSUE TITLE
Option :skip is cool, but how to specify a reason?

### ISSUE THREAD

**Issue by paddor** [rspec/rspec-core#2138]
I have some example groups that I only want to run on JRuby/non-JRuby. So I do this:

```ruby
    context "on non-jruby", skip: RUBY_ENGINE == "jruby" do
      context "on 64-bit system" do
        let(:size) { double("uint64") }
        before(:each) { stub_const "::FFI::Pointer::SIZE", 8 }
        before(:each) { expect(ptr).to receive(:read_uint64).and_return(size) }
        it "reads uint64" do
          # ...
        end
      end
      context "on 32-bit system" do
        let(:size) { double("uint32") }
        before(:each) { stub_const "::FFI::Pointer::SIZE", 4 }
        before(:each) { expect(ptr).to receive(:read_uint32).and_return(size) }
        it "reads uint32" do
          # ...
        end
      end
    end

    context "on jruby", skip: RUBY_ENGINE != "jruby" do
      let(:size) { double("ulong_long") }
      before(:each) { expect(ptr).to receive(:read_ulong_long).and_return(size) }
      it "reads ulong_long" do
        # ...
      end
    end
```

When I run it, I'll see:

```
...
on jruby
    reads ulong_long (PENDING: No reason given)
```
How to specify a reason?

**Comment by JonRowe** [comment]
`skip` works if there is a reason present, or true, (equally false and nil prevent skipping) so you need something like:
```ruby
context "on jruby", skip: (RUBY_ENGINE != "jruby" ? : 'reason' : nil) do
  # ...
end
```

**Comment by paddor** [comment]
Thanks.

**Comment by flatrocks** [comment]
In simpler terms:
``` Ruby
context "some context", skip: "Some reason here" do
   ...
```
Then the "# No reason given" output will be replaced by "# Some reason here"

**Comment by pirj** [comment]
![image](https://user-images.githubusercontent.com/6916/82162355-85836880-98ac-11ea-9838-86534dc58e12.png)

Just in case, it is replaced.
